Source/Core/IO/Logging/InMemoryLogger.cs

```diff
@@ -49,6 +49,14 @@ public InMemoryLogger()
             this.Lock = new object();
         }
 
+        /// <summary>
+        /// Clear acculated log data.
+        /// </summary>
+        public void Clear()
```

Comment:
Instead of introducing a new method for this, could we override the `Dispose` method (the `ILogger` is disposable and the base `TextWriter` implements it) to do this? Then we won't need to change the `TestingEngine` too :)

Reply:
Except that TextWriter did not expose a virtual Dispose method :-(

Reply:
Oh no, you are right, that sucks man :(

Reply:
I guess we can always cheat using `new void Dispose() { base.Dispose() }`, but dunno ... I guess we are the only ones that have access to this logger right? Because if someone only has the TextWriter it can leak without a way to call either `Clear` or the `new Dispose` on the subclass ...

Reply:
Oh, I know, we need to use `Close` which is virtual, which is actually what someone needs to call instead of `Dispose` on a `TextWriter` (it calls `Dispose` internally).